HealthcareSystem/EventSourcingService/Service/PatientSchedulingEventService/EventStorePatientStepSchedulingService.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using EventSourcingService.DTO;
+using EventSourcingService.DTO.PatientSchedulingEventDTO;
+using EventSourcingService.Model;
+using EventSourcingService.Model.Enum;
+using EventSourcingService.Repository;
+
+namespace EventSourcingService.Service
+{
+    public class EventStorePatientStepSchedulingService : IEventStorePatientStepSchedulingService
+    {
+
+        private readonly IDomainEventRepository<PatientStepSchedulingEvent> _patientSchedulingEventRepository;
+
+        public EventStorePatientStepSchedulingService(IDomainEventRepository<PatientStepSchedulingEvent> patientSchedulingEventRepository)
+        {
+            _patientSchedulingEventRepository = patientSchedulingEventRepository;
+        }
+
+        public IEnumerable<PatientStepSchedulingEvent> GetAll()
+        {
+            return _patientSchedulingEventRepository.GetAll();
+        }
+
+        public PatientStepSchedulingEvent Add(PatientStepSchedulingEventDTO patientStepSchedulingEventDTO)
+        {
+            return _patientSchedulingEventRepository.Add(new PatientStepSchedulingEvent()
+            {
+                StartSchedulingEventId = patientStepSchedulingEventDTO.StartSchedulingEventId,
+                UserAge = patientStepSchedulingEventDTO.UserAge,
+                UserGender = patientStepSchedulingEventDTO.UserGender,
+                EventStep = patientStepSchedulingEventDTO.EventStep,
+                ClickEvent = patientStepSchedulingEventDTO.ClickEvent
+            });
+        }
+
+        public StepClosureStatisticDTO ClosedSchedulingStepStatistic()
+        {
+            StepClosureStatisticDTO closedSchedulingStepStatistic = new StepClosureStatisticDTO();
```

Comment:
Why not do a separate per step DTO that contains a string or the enum for the Step name, and then statistics? And then you can just calculate a list of those that contains the objects for every step and make everything easier to work with instead of having this huge method and calculating the minimum manually,